File: mobile/cliente/lib/features/documentos/domain/documento.dart
class Documento {
  const Documento({
    required this.id,
    required this.associacaoId,
    required this.imovelId,
    required this.proprietarioId,
    required this.comodatarioId,
    required this.tipo,
    required this.titulo,
    required this.status,
    this.storagePath,
    this.fileName,
    this.observacao,
    required this.createdAt,
  });

  final String id;
  final String associacaoId;
  final String imovelId;
  final String proprietarioId;
  final String comodatarioId;
  final String tipo;
  final String titulo;
  final String status; // solicitado/enviado/aprovado/rejeitado
  final String? storagePath;
  final String? fileName;
  final String? observacao;
  final DateTime createdAt;

  bool get podeEnviar => status == 'solicitado' || status == 'rejeitado';
  bool get podeBaixar => storagePath != null && storagePath!.isNotEmpty;

  factory Documento.fromJson(Map<String, dynamic> j) => Documento(
    id: j['id'] as String,
    associacaoId: j['associacaoId'] as String,
    imovelId: j['imovelId'] as String,
    proprietarioId: j['proprietarioId'] as String,
    comodatarioId: j['comodatarioId'] as String,
    tipo: j['tipo'] as String,
    titulo: j['titulo'] as String,
    status: (j['status'] as String?) ?? 'solicitado',
    storagePath: j['storagePath'] as String?,
    fileName: j['fileName'] as String?,
    observacao: j['observacao'] as String?,
    createdAt:
        DateTime.tryParse((j['createdAt'] as String?) ?? '') ?? DateTime.now(),
  );
}

File: backend/ms-documento/src/domain/entities/Documento.ts
export type DocumentoStatus = 'solicitado' | 'enviado' | 'aprovado' | 'rejeitado';

export interface Documento {
  id: string;
  associacaoId: string;
  imovelId: string;
  proprietarioId: string;
  comodatarioId: string;
  tipo: string;
  titulo: string;
  status: DocumentoStatus;
  storagePath: string | null;
  observacao: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface DocumentoRow {
  id: string;
  associacao_id: string;
  imovel_id: string;
  proprietario_id: string;
  comodatario_id: string;
  tipo: string;
  titulo: string;
  status: DocumentoStatus;
  storage_path: string | null;
  observacao: string | null;
  created_at: string;
  updated_at: string;
}

export function rowToDocumento(row: DocumentoRow): Documento {
  return {
    id: row.id,
    associacaoId: row.associacao_id,
    imovelId: row.imovel_id,
    proprietarioId: row.proprietario_id,
    comodatarioId: row.comodatario_id,
    tipo: row.tipo,
    titulo: row.titulo,
    status: row.status,
    storagePath: row.storage_path,
    observacao: row.observacao,
    createdAt: row.created_at,
    updatedAt: row.updated_at,
  };
}
